MapboxNavigation/NavigationViewController.swift

```diff
@@ -494,6 +494,9 @@ extension NavigationViewController: NavigationServiceDelegate {
             userHasArrivedAndShouldPreventRerouting {
             mapViewController?.mapView.updateCourseTracking(location: location, animated: true)
         }
+        
```

Comment:
just out of curiosity, what's the internal guideline wrt `for-loops` and closures like `forEach` usage in nav sdk? specifically referring to line 474.

Reply:
My personal preference is for for-in loops unless a function is being passed into forEach as an argument, so sometimes I nitpick about lines like that. Others have a higher tolerance for these forEach calls, but I view it as an unnecessary barrier to readability.